src/snowflake/snowpark/modin/pandas/series.py

```diff
@@ -1535,19 +1535,35 @@ def set_axis(
             copy=copy,
         )
 
-    @series_not_implemented()
-    def unstack(self, level=-1, fill_value=None):  # noqa: PR01, RT01, D200
+    def unstack(
+        self,
+        level: int | str | list = -1,
+        fill_value: int | str | dict = None,
+        sort: bool = True,
+    ):
         """
         Unstack, also known as pivot, Series with MultiIndex to produce DataFrame.
         """
         # TODO: SNOW-1063347: Modin upgrade - modin.pandas.Series functions
         from snowflake.snowpark.modin.pandas.dataframe import DataFrame
 
-        result = DataFrame(
-            query_compiler=self._query_compiler.unstack(level, fill_value)
-        )
+        # We can't unstack a Series object, if we don't have a MultiIndex.
+        if len(self.index.names) > 1:
+            result = DataFrame(
+                query_compiler=self._query_compiler.unstack(
+                    level, fill_value, sort, is_series_output=False
+                )
+            )
+        else:
+            raise ValueError(  # pragma: no cover
```

Comment:
I think we can simply push all implementation details to backend, the multiIndex check with error, and also the drop level implementation etc. That is what we evetually want to get at the end also, where frontend only responsible for basic input parameter check, simple parameter reduction, and final result format (dataframe, series, inplace) handlization

Reply:
I think we should leave the frontend checks as-is since they are raising errors that pandas raises as well, so no need to go to the query compiler. It will continue to be as consistent as possible with open-source Modin -- I got the query counts down by using `self._query_compiler.has_multiindex`

Reply:
The case where we use `droplevel` case is only if the input is a Series, so how do we push it to the query compiler layer?

Reply:
Resolved offline, passing `is_series_input` to the QC layer.